trusona-sdk-http/src/main/java/com/trusona/sdk/http/client/BaseErrorHandler.java

```diff
@@ -4,28 +4,30 @@
 import com.trusona.sdk.http.ErrorHandler;
 import com.trusona.sdk.http.client.v2.response.ErrorResponse;
 import com.trusona.sdk.resources.exception.TrusonaException;
+import okhttp3.ResponseBody;
 import retrofit2.Response;
 
 import java.io.IOException;
-import java.io.Reader;
+import java.util.Optional;
 
 public abstract class BaseErrorHandler implements ErrorHandler {
 
   protected ErrorResponse getErrorResponse(Response response) throws TrusonaException {
-    Reader errorBody = response.errorBody().charStream();
+    Optional<ResponseBody> errorBody = Optional.ofNullable(response.errorBody());
 
-    try {
-      return JacksonConfig.getObjectMapper().readValue(errorBody, ErrorResponse.class);
-    }
-    catch (IOException e) {
-      throw new TrusonaException("An unknown error occurred. Contact Trusona to determine the exact cause.", e);
-    }
-    finally {
+    if (errorBody.isPresent()) {
       try {
-        errorBody.close();
+        return JacksonConfig.getObjectMapper().readValue(errorBody.get().charStream(), ErrorResponse.class);
       }
       catch (IOException e) {
+        throw new TrusonaException("An unknown error occurred. Contact Trusona to determine the exact cause.", e);
```

Comment:
Couldn't we be more specific? The error isn't really unknown, it is that the error body failed to parse...

Reply:
Not new code exactly - i more or less fixed a call where the upstream call clearly specified that the method could return `null`